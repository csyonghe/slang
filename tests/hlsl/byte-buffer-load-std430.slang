//TEST(compute, vulkan):COMPARE_COMPUTE_EX:-vk -compute -shaderobj -output-using-type -xslang -fvk-use-gl-layout

//TEST_INPUT:ubuffer(data=[0], stride=4):out,name=outputBuffer
RWStructuredBuffer<float> outputBuffer;

//TEST_INPUT:set inBuffer = ubuffer(data=[1.0 2.0 3.0 4.0 5.0 6.0 7.0 8.0], stride=4)
RWByteAddressBuffer inBuffer;

struct MyStruct
{
    float v0;
    float2 v1;
    float2 v2;
}

[numthreads(1, 1, 1)]
void computeMain(uint3 dispatchThreadID : SV_DispatchThreadID)
{
    var v = inBuffer.Load<MyStruct>(0);
    outputBuffer[dispatchThreadID.x] = v.v2.x + v.v1.x; // expect 8.0
}
