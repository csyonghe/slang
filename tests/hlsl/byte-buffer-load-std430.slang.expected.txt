type: float
8.000000
